edn/src/symbols.rs

```diff
@@ -98,12 +101,14 @@ impl Keyword {
 }
 
 impl NamespacedKeyword {
-    pub fn new(namespace: &str, name: &str) -> Self {
-        assert!(!name.is_empty(), "Keywords cannot be unnamed.");
-        assert!(!namespace.is_empty(), "Keywords cannot have an empty non-null namespace.");
+    pub fn new<T>(namespace: T, name: T) -> Self where T: Into<String> {
```

Comment:
This patch isn't related to reversing, but handling reverse in #212 should be handled differently because of the above comments.